src/wp-includes/cache-compat.php

```diff
@@ -141,3 +141,29 @@ function wp_cache_flush_runtime() {
 		return wp_using_ext_object_cache() ? false : wp_cache_flush();
 	}
 endif;
+
+if ( ! function_exists( 'wp_cache_flush_group' ) ) :
+	/**
+	 * Removes all cache items in a group.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @see WP_Object_Cache::flush_group()
+	 * @global WP_Object_Cache $wp_object_cache Object cache global instance.
+	 *
+	 * @param string|array $group name(s) of group to remove from cache.
+	 *
+	 * @return bool|Array True or array of bool in array passed on success, false on failure group not found.
+	 */
+	function wp_cache_flush_group( $group ) {
+		global $wp_object_cache;
+
+		if ( $wp_object_cache && method_exists( $wp_object_cache, 'flush_group' ) ) {
+			return $wp_object_cache->flush_group( $group );
+		}
+
+		_doing_it_wrong( __FUNCTION__, __( 'Your implementation of object caching does not support flush by group.' ), '6.0.0' );
+
+		return wp_cache_flush();
```

Comment:
This is a dangerous fallback for this method, having plugin code expecting to flush a specific cache group(s) and then flushing the entire cache is not going to end well.

Additionally, IMHO it's not a `_doing_it_wrong()` `E_USER_DEPRECATED`, possibly a `E_USER_NOTICE` level warning instead though?

Reply:
The worry here, is that if you call this function can expect and clear your group and it doesn't we need to warn developer this is happening. Maybe we should return a `WP_Error` object?

So you could do this in your own code. 
```
$result = wp_cache_flush_group( 'posts' );
if( is_wp_error( $result ) ) {
    wp_cache_flush();
}
```

But give devs more control here.

Reply:
I like the `WP_Error` approach.

Reply:
I think `return false` would suffice myself if it wasn't going to flush everything.. "Clear this cache group, sorry, that failed.".

But as mentioned in https://github.com/WordPress/wordpress-develop/pull/2368#issuecomment-1055750476 variants whose cache server that don't support clearing by prefix, could rotate a prefix to flush the group.

Reply:
So that means that false, could mean a lot of things. Unsupported and the cache group flush failed. That is unclear for developers that using this function, why the failure.

Reply:
Some feedback from @felixarntz:

>What I would suggest is considering making support of this more explicitly optional. For example
> - Not put the `flush_group()` method in the base class, or alternatively put it in the base class as a no-op that just returns `false`.
>- Introduce `wp_cache_flush_group()`, but explicitly document it that it may not be supported and therefore returns a boolean to indicate success, which is critical to check here (other than maybe on other cache functions).
>- In the function, depending on my first point, either check if the method exists on the cache implementation and if not return `false`, or otherwise just call the method and return through its boolean result.
>
>Alternatively, we could make it even more explicit to clarify that developers need to be aware that this may not work, e.g. by having the boolean function just as a check that has to be explicitly called (e.g. `wp_cache_can_flush_group()`) before calling `wp_cache_flush_group()`. The latter could cause a `_doing_it_wrong()` if not supported or throw an exception.
>
>So TLDR whichever way we go, I definitely think it should be explicitly built and marked as optional.

Reply:
@felixarntz's comments here reflect statement I have made throughout this code review. These needs to be a way for developers to know if running this function is going to work. They have to know if the object cache implement supports it. 

I recommend adding a const, like `WP_CACHE_SUPPORTS_GROUP_FLUSH`, so developers of object cache drop-ins can "opt-in" to support this feature. If this const is not defined, then, developers know not to use this function. 

See these simplier [examples](https://github.com/pantheon-systems/wp-redis/blob/b2e6b228a7fe379a9f97cac4453fc4b1792be478/object-cache.php#L16-L26).

Reply:
I'd suggest is using `WP_OBJECTCACHE_*` since `WP_CACHE` refers to WP's "page caching" mechanism.

Reply:
I left a half thought in a Slack thread that I'll leave in full detail here so it doesn't get lost. 

For a lot of groups currently, WP uses the `last_changed` cache key to determine when the group was last changed. It acts as a faux group flushing mechanism. 

https://github.com/WordPress/wordpress-develop/blob/8277055e37c19cd69cbf829478004b3445c8bb79/src/wp-includes/post.php#L7406

For drop-ins that support the `flush_group()` method, WP could call that and trust the drop in developer to have checked their persistent cache supports it.

For drop-ins that do not include the method, WP could include something similar to polyfill the feature. 

```php
// Within caches
if ( ! method_exists( ... )  && 'wp_group_suffiixes' !== $group ) {
   $group = $group . '::' . wp_cahce_group_suffix( $group );
}

// Function
function wp_cahce_group_suffix( $group, $update = false ) {
  $suffix = wp_cache_get( $group, 'wp_group_suffiixes' );
  if ( ! $update ) {
    return $suffix;
  }
  
  $new_suffix = $suffix;
  while ( $new_suffix === $suffix ) {
     $new_suffix = wp_generate_password( 8, true, true );
  }

  return $new_suffix;
}
```

At this stage, I'd call this a thought bubble and it absolutely needs further thinking through. One of the issues Dion mentioned in Slack is that there is potential to hit key length issues in some persistent caches.

It's also possible this is a terrible idea.

Reply:
I think the discussion for this ticket should move back to the original tickets. GH is a code review tool and not best designed for this.